src/core/sys/posix/spawn.d

```diff
@@ -0,0 +1,319 @@
+}
+
+int posix_spawnattr_getsigdefault(const posix_spawnattr_t*, sigset_t*);
+int posix_spawnattr_getsigmask(const posix_spawnattr_t*, sigset_t*);
+int posix_spawnattr_init(posix_spawnattr_t*);
+int posix_spawnattr_setflags(posix_spawnattr_t*, short);
+int posix_spawnattr_setpgroup(posix_spawnattr_t*, pid_t);
+int posix_spawnattr_setsigdefault(posix_spawnattr_t*, const sigset_t*);
+int posix_spawnattr_setsigmask(posix_spawnattr_t*, const sigset_t*);
+int posix_spawn(pid_t*pid, const char* path,
+                const posix_spawn_file_actions_t* file_actions,
+                const posix_spawnattr_t* attrp,
+                const char** argv, const char** envp);
+int posix_spawnp(pid_t* pid, const char* file,
+                 const posix_spawn_file_actions_t* file_actions,
+                 const posix_spawnattr_t* attrp,
+                 const char** argv, const char** envp);
+
+version (linux)
+{
+    version (CRuntime_Glibc)
+    {
+        // Source: https://sourceware.org/git/?p=glibc.git;a=blob;f=posix/spawn.h;hb=HEAD
+        enum
+        {
+            POSIX_SPAWN_RESETIDS = 0x01,
+            POSIX_SPAWN_SETPGROUP = 0x02,
+            POSIX_SPAWN_SETSIGDEF = 0x04,
+            POSIX_SPAWN_SETSIGMASK = 0x08,
+            POSIX_SPAWN_SETSCHEDPARAM = 0x10,
+            POSIX_SPAWN_SETSCHEDULER = 0x20
+        }
+        import core.sys.posix.config : __USE_GNU;
+        static if (__USE_GNU)
+        {
+            enum
+            {
+                POSIX_SPAWN_USEVFORK = 0x40,
+                POSIX_SPAWN_SETSID = 0x80
+            }
+        }
+        struct posix_spawnattr_t
+        {
+            short __flags;
+            pid_t __pgrp;
+            sigset_t __sd;
+            sigset_t __ss;
+            sched_param __sp;
+            int __policy;
+            int[16] __pad;
+        }
+        struct __spawn_action;
+        struct posix_spawn_file_actions_t
+        {
+            int __allocated;
+            int __used;
+            __spawn_action* __actions;
+            int[16] __pad;
+        }
+    }
+    version (CRuntime_Bionic)
+    {
+        // Source: https://android.googlesource.com/platform/bionic.git/+/master/libc/include/spawn.h
+        enum
+        {
+            POSIX_SPAWN_RESETIDS = 1,
+            POSIX_SPAWN_SETPGROUP = 2,
+            POSIX_SPAWN_SETSIGDEF = 4,
+            POSIX_SPAWN_SETSIGMASK = 8,
+            POSIX_SPAWN_SETSCHEDPARAM = 16,
+            POSIX_SPAWN_SETSCHEDULER = 32
+        }
+        import core.sys.posix.config : __USE_GNU;
+        static if (__USE_GNU)
+        {
+            enum
+            {
+                POSIX_SPAWN_USEVFORK = 64,
+                POSIX_SPAWN_SETSID = 128
+            }
+        }
+        alias posix_spawnattr_t = __posix_spawnattr*;
+        alias posix_spawn_file_actions_t = __posix_spawn_file_actions*;
+        struct __posix_spawnattr;
+        struct __posix_spawn_file_actions;
+    }
+    version (CRuntime_Musl)
```

Comment:
For consistency could this be `else version`? Same above for `CRuntime_Bionic`.

Reply:
Sure, thanks for pointing this out, don't know how I missed it.